specification/network/resource-manager/Microsoft.Network/stable/2023-05-01/virtualNetwork.json

```diff
@@ -1633,7 +1633,15 @@
           "description": "Application gateway IP configurations of virtual network resource."
         },
         "defaultOutboundAccess": {
-          "type": "boolean",
+          "type": "string",
+          "enum": [
```

Comment:
Introducing boolean enums true/false is an antipattern. Why do you want to change it? Do you really want to model it as something with more than two possible states, to make a future proof API?

Reply:
@TimLovellSmith I got comment from we should use Enum instead of Boolean. Also, we may have more values in future for this property.

https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md

Reply:
If you think you'll have future values, then 'true', 'false' don't make sense. You need to redesign it, e.g.

'outboundAccessRule': 'DefaultAllow', 'DefaultDeny', ...add more in future

Reply:
Thanks @TimLovellSmith for suggestion. I discussed with developer who added this property as NRP side, and we don't expect multiple values for this specific property. So, changed back to Boolean.